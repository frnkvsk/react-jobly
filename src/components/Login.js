import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';
import { Container, FormHelperText, } from '@material-ui/core';
import OutlinedInput from '@material-ui/core/OutlinedInput';
import { Button } from '@material-ui/core';

import {login} from '../api/JoblyApi'
import {useFormInput} from '../hooks/useFormInput';

function LinkTab(props) {
  return (
    <Tab
      component="a"
      onClick={(event) => {
        event.preventDefault();
      }}
      {...props}
    />
  );
}

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    maxWidth: '500px',
    padding: '30px',
    height: '100vh',
    // backgroundColor: '#ffffff',
  },
  form: {
    // border: '1px solid red',
    width: '100%',
    backgroundColor: '#ffffff',
  },
  label: {
    fontSize: '18px',
    fontWeight: 'bold',
    width: '90%',
    margin: '10px 20px 0 20px',
  },
  input: {
    width: '90%',
    margin: '0 20px 0 20px',
  },
  button: {
    display: 'flex',
    justifyContent: 'flex-end',
    margin: '10px 20px 15px',
  }

}));

export default function Login(login_status) {
  

  const classes = useStyles();
  const [loginType, setValue] = React.useState(login_status);

  const username = useFormInput("");
  const password = useFormInput("");
  const first_name = useFormInput("");
  const last_name = useFormInput("");
  const email = useFormInput("");
  const handleChange = (e, newValue) => {
    setValue(newValue);
  };
  const handleSubmit = () => {
    login(username.value, password.value);
  }
  return (
    
    <Container className={classes.root}>
      <AppBar position="static">
        <Tabs
          variant="fullWidth"
          value={loginType}
          onChange={handleChange}
          aria-label="nav tabs example"
        >
          <LinkTab label="Login" value="login" />
          <LinkTab label="Sign up" value="signup" />
        </Tabs>
      </AppBar>
      
      {loginType === 'login' ?    
        <form className={classes.form} noValidate autoComplete="off">
          <FormHelperText className={classes.label} id="standard-weight-helper-text">username</FormHelperText>
          <OutlinedInput className={classes.input} name="username" variant="outlined" {...username}/>

          <FormHelperText className={classes.label} id="standard-weight-helper-text">Password</FormHelperText>
          <OutlinedInput className={classes.input} name="password" type="password" variant="outlined" {...password} />
          <div className={classes.button}>
            <Button variant="contained" color="primary" onClick={handleSubmit}>Submit</Button>
          </div>
        </form>
                :
        <form className={classes.form}>      
          <FormHelperText className={classes.label} id="standard-weight-helper-text">username</FormHelperText>
          <OutlinedInput className={classes.input} name="username" variant="outlined" {...username}/>

          <FormHelperText className={classes.label} id="standard-weight-helper-text">Password</FormHelperText>
          <OutlinedInput className={classes.input} name="password" type="password" variant="outlined" {...password}/>

          <FormHelperText className={classes.label} id="standard-weight-helper-text">First Name</FormHelperText>
          <OutlinedInput className={classes.input} name="first_name" variant="outlined" {...first_name}/>

          <FormHelperText className={classes.label} id="standard-weight-helper-text">Last Name</FormHelperText>
          <OutlinedInput className={classes.input} name="last_name" variant="outlined" {...last_name}/>

          <FormHelperText className={classes.label} id="standard-weight-helper-text">Email</FormHelperText>
          <OutlinedInput className={classes.input} name="email" variant="outlined" {...email}/>
          <div className={classes.button}>
            <Button variant="contained" color="primary" onClick={handleSubmit}>Submit</Button>
          </div>
          
        </form>
      }
    </Container>
  );
}


