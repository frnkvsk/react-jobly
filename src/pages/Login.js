/** Login and Signup */
import React, { useState, useContext } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';
import { Container, FormHelperText, Box, Button } from '@material-ui/core';
import OutlinedInput from '@material-ui/core/OutlinedInput';
import { useHistory } from "react-router-dom";
import { login, signup, getUserInfo } from '../api/JoblyApi'
import { useFormInput } from '../hooks/useFormInput';
import { AuthContext } from '../context/AuthContext';

function LinkTab(props) {
  return (
    <Tab
      component="a"
      onClick={(event) => {
        event.preventDefault();
      }}
      {...props}
    />
  );
}

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    maxWidth: '500px',
    padding: '30px',
    height: '100vh',
  },
  form: {
    width: '100%',
    backgroundColor: '#ffffff',
  },
  label: {
    fontSize: '18px',
    fontWeight: 'bold',
    width: '90%',
    margin: '10px 20px 0 20px',
  },
  input: {
    width: '90%',
    margin: '0 20px 0 20px',
  },
  button: {
    display: 'flex',
    justifyContent: 'flex-end',
    margin: '10px 20px 15px',
  },
  err: {
    color: '#ff1744',
    fontSize: '24px',
  }

}));

export default function Login() {
  const classes = useStyles();
  const history = useHistory();
  const auth = useContext(AuthContext);
  const [loginType, setValue] = useState("signup");
  const [errorMessage, setErrorMessage] = useState(false);  
  const username = useFormInput("");
  const password = useFormInput("");
  const first_name = useFormInput("");
  const last_name = useFormInput("");  
  const photo_url = useFormInput("");
  const email = useFormInput("");
  const handleChange = (e, newValue) => {
    setValue(newValue);
  };
  const handleSubmitLogin = async () => {
    setErrorMessage(false);

    try {
      // verify username and password are correct
      const resp = await login(username.value, password.value);

      // if logged in, use resp.token to get user information
      const userInfo = await getUserInfo(username.value, resp.token);

      auth.setAuthState({
        userInfo: userInfo.user,
        token: resp.token,
      });
      history.push(`/`);
    } catch (error) {
      setErrorMessage(true);
    }    
  }
  const handleSubmitSignup = async () => {
    setErrorMessage(false);
    try {
      const resp = await signup(
          username.value, 
          password.value, 
          first_name.value, 
          last_name.value, 
          photo_url.value, 
          email.value);
      auth.setAuthState({
        token: resp.token,
        userInfo: {
          username: username.value,
          first_name: first_name.value,
          last_name: last_name.value,
          photo_url: photo_url.value,
          email: email.value,
        }
      });
      history.push(`/`);
    } catch (error) {
      setErrorMessage(true);
    }    
  }

  return (
    
    <Container className={classes.root}>
      <Box className={classes.err} 
        component="span" 
        display={errorMessage ? 'block' : 'none'}
        >
          Error: Invalid credentials
      </Box>
      <AppBar position="static">
        <Tabs
          variant="fullWidth"
          value={loginType}
          onChange={handleChange}
          aria-label="nav tabs example"
        >
          <LinkTab label="Login" value="login" />
          <LinkTab label="Sign up" value="signup" />
        </Tabs>
      </AppBar>
      
      {loginType === 'login' ?    
        <form className={classes.form}>
          <FormHelperText className={classes.label} >username</FormHelperText>
          <OutlinedInput className={classes.input} name="username" variant="outlined" {...username}/>

          <FormHelperText className={classes.label} >Password</FormHelperText>
          <OutlinedInput className={classes.input} name="password" type="password" variant="outlined" {...password} />
          <div className={classes.button}>
            <Button variant="contained" color="primary" onClick={handleSubmitLogin}>Submit</Button>
          </div>
        </form>
                :
        <form className={classes.form}>      
          <FormHelperText className={classes.label} >username</FormHelperText>
          <OutlinedInput className={classes.input} name="username" variant="outlined" {...username}/>

          <FormHelperText className={classes.label} >Password</FormHelperText>
          <OutlinedInput className={classes.input} name="password" type="password" variant="outlined" {...password}/>

          <FormHelperText className={classes.label} >First Name</FormHelperText>
          <OutlinedInput className={classes.input} name="first_name" variant="outlined" {...first_name}/>

          <FormHelperText className={classes.label} >Last Name</FormHelperText>
          <OutlinedInput className={classes.input} name="last_name" variant="outlined" {...last_name}/>

          <FormHelperText className={classes.label} >Photo URL</FormHelperText>
          <OutlinedInput className={classes.input} name="photo_url" variant="outlined" {...photo_url}/>

          <FormHelperText className={classes.label} >Email</FormHelperText>
          <OutlinedInput className={classes.input} name="email" variant="outlined" {...email}/>
          <div className={classes.button}>
            <Button variant="contained" color="primary" onClick={handleSubmitSignup}>Submit</Button>
          </div>
          
        </form>
      }
    </Container>
  );
}


