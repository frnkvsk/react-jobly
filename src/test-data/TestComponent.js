// import React, { useContext } from 'react';
// import { AuthContext } from '../context/AuthContext';

// const TestComponent = () => {
//   const auth = useContext(AuthContext);
//   const setAuth2 = async () =>
//     await auth.setAuthState({
//       token: 'testtoken',
//       userInfo: {
//       username: 'testuser',
//       first_name: 'testfirst',
//       last_name: 'testlast',
//       email: 'test@test.com',
//       photo_url: null
//     }});
//   (async () => await setAuth2())();
//   console.log('-------auth.authState',auth.authState)
//   return (
//     <div>
//       <div>{auth.authState.userInfo.username}</div>
//       <div>{auth.authState.userInfo.first_name}</div>
//       <div>{auth.authState.userInfo.last_name}</div>
//       <div>{auth.authState.userInfo.email}</div>
//       <div>{auth.authState.userInfo.photo_url}</div>
//     </div>
//   )
// }
// export default TestComponent;

