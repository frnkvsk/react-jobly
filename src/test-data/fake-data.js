const testData = {
  jobs: [
    {
    company_handle: "test-company-a",
    equity: 0.115,
    id: 1,
    salary: 11111,
    state: null,
    title: "Editor, magazine features",
    },
      {
      company_handle: "test-company-b",
      equity: 0.22,
      id: 2,
      salary: 22222,
      state: null,
      title: "Coder",
    },
      {
      company_handle: "test-company-c",
      equity: 0.33,
      id: 3,
      salary: 33333,
      state: null,
      title: "Front-end Dev",
    },
  ],
  companies: [
    {
      description: "Test description 1",
      handle: "test-company-1",
      logo_url: "http://www.gtdesigns.it/wp-content/uploads/OverusedLogos/99gen_circle.jpg",
      name: "Test Company 1",
    },
    {
      description: "Test description 2",
      handle: "test-company-2",
      logo_url: "http://www.gtdesigns.it/wp-content/uploads/OverusedLogos/99gen_circle.jpg",
      name: "Test Company 2",
    },
    {
      description: "Test description 3",
      handle: "test-company-3",
      logo_url: "http://www.gtdesigns.it/wp-content/uploads/OverusedLogos/99gen_circle.jpg",
      name: "Test Company 3",
    },
  ],
}

export { testData }