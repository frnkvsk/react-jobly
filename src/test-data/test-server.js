import { rest } from 'msw';
import { setupServer } from 'msw/node';

let server;

const setServer = async (url, status, message='') => {
  if(server) {
    server.close();
    server.resetHandlers();
  }
  server = await setupServer(
    rest.get(url, (req, res, ctx) => {
      return res(
        ctx.status(status),
        ctx.json(message)
      )
    })
  );
}
  

beforeAll(() => server ? server.listen() : null);
afterAll(() => server ? server.close() : null);
afterEach(() => server ? server.resetHandlers() : null);

export { setServer, server, rest }
