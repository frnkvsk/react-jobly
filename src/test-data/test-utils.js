// test-utils.js
import React from 'react';
import { render } from '@testing-library/react';

import { BrowserRouter } from 'react-router-dom';
import { SearchProvider } from '../context/SearchContext';
import { AuthProvider } from '../context/AuthContext';
import { PageCountProvider } from '../context/PageCountContext';

const AllTheProviders = (page) => {
  return render(
    <BrowserRouter>
      <AuthProvider>
        <SearchProvider>
          <PageCountProvider>
            {page}
          </PageCountProvider>
        </SearchProvider>
      </AuthProvider> 
    </BrowserRouter>
    
  );
}

// re-export everything
export * from '@testing-library/react';

// override render method
export { AllTheProviders as render }
