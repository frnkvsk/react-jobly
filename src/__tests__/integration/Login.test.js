import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, screen, fireEvent } from '../../test-data/test-utils';
import Login from '../../pages/Login';

// smoke and appearance tests
test('renders Login ', async() => {
  render(<Login />);
  
  expect(screen.getByText(/username/i)).toBeInTheDocument();

  expect(screen.getByText(/password/i)).toBeInTheDocument();

  // click the 'SIGN UP' tab to load the sign up form
  fireEvent.click(screen.getByRole('tab', {name: /sign up/i}));

  // test if sign up form loaded
  expect(screen.queryByText(/first name/i)).toBeInTheDocument();

  // click the 'LOGIN' tab to load the login form
  fireEvent.click(screen.getByRole('tab', {name: /login/i}));

  // test if login form loaded
  expect(screen.queryByText(/first name/i)).not.toBeInTheDocument();

  expect(screen.getByRole('button', {name: /submit/i})).toBeInTheDocument();
});


// snapshot test
// test('matches snapshot', () => {
//   const { asFragment } = render(<Login />);
//   expect(asFragment()).toMatchSnapshot();
// });