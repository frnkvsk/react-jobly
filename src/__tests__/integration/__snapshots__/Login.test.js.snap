// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root makeStyles-root-16 MuiContainer-maxWidthLg"
  >
    <span
      class="MuiBox-root MuiBox-root-22 makeStyles-err-21"
    >
      Error: Invalid credentials
    </span>
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        class="MuiTabs-root"
      >
        <div
          class="MuiTabs-scroller MuiTabs-fixed"
          style="overflow: hidden;"
        >
          <div
            aria-label="nav tabs example"
            class="MuiTabs-flexContainer"
            role="tablist"
          >
            <a
              aria-disabled="false"
              aria-selected="true"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-selected MuiTab-fullWidth"
              role="tab"
              tabindex="0"
            >
              <span
                class="MuiTab-wrapper"
              >
                Login
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </a>
            <a
              aria-disabled="false"
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit MuiTab-fullWidth"
              role="tab"
              tabindex="-1"
            >
              <span
                class="MuiTab-wrapper"
              >
                Sign up
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </a>
          </div>
          <span
            class="PrivateTabIndicator-root-23 PrivateTabIndicator-colorSecondary-25 MuiTabs-indicator"
            style="left: 0px; width: 0px;"
          />
        </div>
      </div>
    </header>
    <form
      class="makeStyles-form-17"
    >
      <p
        class="MuiFormHelperText-root makeStyles-label-18"
      >
        Username
      </p>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root makeStyles-input-19"
        variant="outlined"
      >
        <input
          class="MuiInputBase-input MuiOutlinedInput-input"
          name="username"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-27 MuiOutlinedInput-notchedOutline"
          style="padding-left: 8px;"
        >
          <legend
            class="PrivateNotchedOutline-legend-28"
            style="width: 0.01px;"
          >
            <span>
              ​
            </span>
          </legend>
        </fieldset>
      </div>
      <p
        class="MuiFormHelperText-root makeStyles-label-18"
      >
        Password
      </p>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root makeStyles-input-19"
        variant="outlined"
      >
        <input
          class="MuiInputBase-input MuiOutlinedInput-input"
          name="password"
          type="password"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-27 MuiOutlinedInput-notchedOutline"
          style="padding-left: 8px;"
        >
          <legend
            class="PrivateNotchedOutline-legend-28"
            style="width: 0.01px;"
          >
            <span>
              ​
            </span>
          </legend>
        </fieldset>
      </div>
      <div
        class="makeStyles-button-20"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Submit
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
